scripts/download_file.py

```diff
@@ -0,0 +1,156 @@
+
+import argparse
+import urllib.request
+import urllib.error
+import os
+import hashlib
+import sys
+
+
+parser = argparse.ArgumentParser(description="Download data file for EFI KBase app")
+parser.add_argument("--remote-dir", required=True, type=str, help="URL of the directory in which the file resides")
+parser.add_argument("--remote-file", required=True, type=str, help="Name of the file on the remote to download")
+parser.add_argument("--local-dir", required=True, type=str, help="Path to local directory where file will be unpacked")
+parser.add_argument("--local-file", required=True, type=str, help="Path to file on local system where download file is placed")
+args = parser.parse_args()
+
+
+if not os.path.isdir(args.local_dir):
+    os.mkdir(args.local_dir)
+
+
+
+
+
+
+def get_remote_file_contents(remote_file):
+    """
+    Downloads the file as a string and return to user; if HTTP error then returns False
+    """
+    try:
+        response = urllib.request.urlopen(remote_file)
+    except urllib.error.HTTPError as e:
+        print(f"Unable to urlopen {remote_file}: {e}")
+        return False
+    file_contents = response.read()
+    file_contents = str(file_contents.decode('utf-8'))
+    return file_contents
+
+
+def get_file_list(args):
+    """
+    Retrieves a list of files on the remote that need to be fetched
+    """
+    remote_file = args.remote_dir + "/" + args.remote_file + ".file_list"
+
+    file_contents = get_remote_file_contents(remote_file)
+    if not file_contents:
+        return [], {} 
+
+    lines = file_contents.splitlines()
+    checksums = {}
+    files = []
+    for line in lines:
+        parts = line.split(" *")
+        files.append(parts[1])
+        checksums[parts[1]] = parts[0]
+
+    return files, checksums
+
```

Comment:
`splitlines` will return a copy of the data. It would be more memory efficient to iterate over the decoded buffer directly. See if you can make some changes in `get_remote_file_contents` so that you get back a [`StringIO`](https://docs.python.org/3/library/io.html#text-i-o) object. Also, it looks like you maintain the list of filenames both in an array and as the keys of the dict. Could you just use the dict and call `.keys()` when the filenames are needed?

Reply:
The data returned by `get_remote_file_contents` will be quite small.  It is simply the output of running `md5sum dir/files* > dir/dir.file_list`.  However, I think what you suggest is the correct approach.  I will investigate.  Regarding the array/dict, the files need to be concatenated in order.  I believe that dictionaries in Python are ordered so yes, I can return one dict rather than an array and dict.

Reply:
@1ndy The default Python install on some OS is Python 3.6 and implicitly ordered dictionaries were introduced after 3.6.  Do you think its safe to require a specific version of Python in order for the script to run?  (By using `sys.version_info` for example.)  It looks like KBase itself is based on a Python 3.8 implementation.

Reply:
Ah right, yeah I wouldn't try to force a specific python version just for this feature, you can use [`OrderedDict`](https://docs.python.org/3/library/collections.html#collections.OrderedDict) instead